spec/requests/vendor_api/v1.1/post_confirm_deferred_offer_spec.rb

```diff
@@ -0,0 +1,138 @@
+    create(:submitted_application_choice,
+           :with_completed_application_form,
+           application_trait,
+           course_option: original_course_option,
+           current_course_option: original_course_option)
+  end
+  let(:original_course) do
+    create(:course,
+           :open_on_apply,
+           :previous_year_but_still_available,
+           provider: currently_authenticated_provider)
+  end
+  let(:original_course_option) do
+    create(:course_option,
+           :previous_year_but_still_available,
+           course: original_course)
+  end
+
+  it_behaves_like 'an endpoint that requires metadata', '/confirm-deferred-offer', '1.1'
+
+  describe 'request body' do
+    context 'when valid' do
+      let(:request_body) { { data: { conditions_met: false } } }
+
+      it 'has conditions_met set' do
+        expect(request_body[:data]).to be_valid_against_openapi_schema('ConfirmDeferredOffer', '1.1')
+      end
+    end
+
+    context 'when invalid' do
+      it 'when `data` is missing from the request_body it renders an error' do
+        post_api_request "/api/v1.1/applications/#{application_choice.id}/confirm-deferred-offer", params: { data: {} }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(parsed_response['errors'].map { |error| error['error'] })
+          .to contain_exactly('ParameterMissing')
+        expect(parsed_response['errors'].map { |error| error['message'] })
+          .to contain_exactly('param is missing or the value is empty: data')
+      end
+
+      it 'when `conditions_met` is missing from the request_body it renders an error' do
+        post_api_request "/api/v1.1/applications/#{application_choice.id}/confirm-deferred-offer", params: { data: { any_param: '' } }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(parsed_response['errors'].map { |error| error['error'] })
+          .to contain_exactly('ParameterMissing')
+        expect(parsed_response['errors'].map { |error| error['message'] })
+          .to contain_exactly('param is missing or the value is empty: conditions_met')
+      end
+    end
+  end
+
+  describe 'confirming a deffered offer' do
+    context 'when the application is not in a state that allows offer deferal' do
+      let(:application_trait) { :with_offer }
+
+      it 'renders an Unprocessable Entity error' do
+        post_api_request "/api/v1.1/applications/#{application_choice.id}/confirm-deferred-offer", params: request_body
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(parsed_response).to be_valid_against_openapi_schema('UnprocessableEntityResponse')
+        expect(parsed_response['errors'].map { |error| error['message'] })
+          .to contain_exactly("It's not possible to perform this action while the application is in its current state")
+      end
+    end
+
+    context 'when the offered course does not exist in the new cycle' do
```

Comment:
could we add a spec for when we try to confirm a deferred offer in the same cycle?

Reply:
Spec added in latest commit.